client/my-sites/checkout/checkout-system-decider.js

```diff
@@ -124,34 +125,51 @@ export default function CheckoutSystemDecider( {
 	}
 
 	if ( 'composite-checkout' === checkoutVariant ) {
+		isLoggedOutCart &&
+			initGoogleRecaptcha(
```

Comment:
A couple points about this call to `initGoogleRecaptcha()`

Firstly it's running a side-effect in the middle of the render function, which I think we're getting away with because this component probably doesn't re-render. But I think ideally we'd only want this to run on-mount. So something like:

```typescript
useEffect( () => {
	// Doing the check inside the effect function since we're not allowed to have hooks inside branches
	if ( 'composite-checkout' === checkoutVariant && isLoggedOutCart ) {
		initGoogleRecaptcha( ... );
	}
}, [] ); // `[]` to only run once
```

The other thing is the `'calypso/signup/pageLoad'` event name. I'm not sure what we're ultimately going to use these checkout recaptcha events for, but they seem distinct from the signup ones and we should probably use a different event name. I guess `'calypso/checkout-system-decider/pageLoad'` or something like that?

Reply:
Agreed. Changes updated as suggested.